File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            if (numbers[i] % 2 == 0) {
                answer[i] = numbers[i] + 1;
            } else {
                StringBuilder binary = new StringBuilder(Long.toBinaryString(numbers[i]));
                int lastIndexOf01 = binary.lastIndexOf("01");
                if (lastIndexOf01 >= 0) {
                    binary.setCharAt(lastIndexOf01, '1');
                    binary.setCharAt(lastIndexOf01 + 1, '0');
                } else { //"01"이 없는경우
                    binary.deleteCharAt(0);
                    binary.insert(0, "10");
                }
                answer[i] = Long.valueOf(binary.toString(), 2);
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def solution(numbers):
    answer = []
    for num in numbers:
        if num % 2 == 0:
            answer.append(num + 1)
        else:
            binary = bin(num)[2:]
            last_index_of_01 = binary.rfind("01")
            if last_index_of_01 >= 0:
                binary = binary[:last_index_of_01] + '10' + binary[last_index_of_01 + 2:]
            else:
                binary = '10' + binary[1:]
            answer.append(int(binary, 2))
    return answer